### PR TITLE
feat: add safe effect helper

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- feat: add safeEffect as a saga helper
- feat: wrap registerNanoContract effect on safeEffect

Issue: https://github.com/HathorNetwork/hathor-wallet-mobile/issues/550

>[!NOTE]
>This PR can be cherry-picked to [refactor/default-network](https://github.com/HathorNetwork/hathor-wallet-mobile/tree/refactor/default-network) branch.

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/sagas/helpers.js [hunk 1]
```diff
@@ -344,3 +344,24 @@ export function* progressiveRetryRequest(request, maxRetries = MAX_RETRIES) {
...
+ *
+ * @param {Object} effect The targeted effect.
+ * @param {(error) => void} onError The error handling effect,
+ *    which receives the error object as first argument.
+ *
+ * @returns An anonymous effect.
+ */
+export function safeEffect(effect, onError) {
+  // eslint-disable-next-line func-names
+  return function* (payload) {
```
Comment (tuliomir) [inline src/sagas/helpers.js:360]: question(non-blocking): Why not name this function? Wouldn't it help in an eventual debugging session?

File: src/sagas/nanoContract.js [hunk 2]
```diff
@@ -158,6 +158,16 @@ export function* registerNanoContract({ payload }) {
...
   yield put(nanoContractRegisterSuccess({ entryKey: ncId, entryValue: nc, hasFeedback: true }));
 }
+/**
+ * Effect invoked by safeEffect if an unexpected error occurs.
+ *
+ * @param {Object} error The error captured.
+ */
+function* registerNanoContractOnError(error) {
+  log.error('Unexpected error while registering Nano Contract.', error);
+  yield put(nanoContractRegisterFailure(failureMessage.blueprintInfoFailure));
```
Comment (tuliomir) [inline src/sagas/nanoContract.js:168]: question: Why use the `blueprintInfo` failure when the error was on a registration event?

File: src/sagas/nanoContract.js [hunk 3]
```diff
@@ -440,7 +450,10 @@ export function* requestBlueprintInfo({ payload }) {
 export function* saga() {
   yield all([
     debounce(500, [[types.START_WALLET_SUCCESS, types.NANOCONTRACT_INIT]], init),
-    takeEvery(types.NANOCONTRACT_REGISTER_REQUEST, registerNanoContract),
+    takeEvery(
+      types.NANOCONTRACT_REGISTER_REQUEST,
+      safeEffect(registerNanoContract, registerNanoContractOnError)
```
Comment (tuliomir) [inline src/sagas/nanoContract.js:455]: :star: Nice solution! For the future, I'm thinking of a way to make it more generic, and not require two handler functions for each effect.

**Review by tuliomir — APPROVED** [review]

File: src/sagas/nanoContract.js (hunk 3 above)
Comment (alexruzenhack) [reply src/sagas/nanoContract.js:455]: It would be awesome!

File: src/sagas/nanoContract.js (hunk 2 above)
Comment (alexruzenhack) [reply src/sagas/nanoContract.js:168]: Fixed: 4fb72eb794ed26372460d5700c09d357284141bb

File: src/sagas/helpers.js (hunk 1 above)
Comment (alexruzenhack) [reply src/sagas/helpers.js:360]: I'm giving it a name `_safeEffect`: 554868188bf4f0fc3418d82d7eb12c1153c50298. Thank you!

**Review by tuliomir — APPROVED** [review]

**Pull request merged**